File: holoken-core/src/main/java/com/holokenmod/options/DifficultySetting.java
package com.holokenmod.options;

public enum DifficultySetting {
	ANY(null),
	VERY_EASY(GameDifficulty.VERY_EASY),
	EASY(GameDifficulty.EASY),
	MEDIUM(GameDifficulty.MEDIUM),
	HARD(GameDifficulty.HARD),
	EXTREME(GameDifficulty.EXTREME);
	
	private final GameDifficulty gameDifficulty;
	
	DifficultySetting(GameDifficulty gameDifficulty) {
		this.gameDifficulty = gameDifficulty;
	}
	
	public GameDifficulty getGameDifficulty() {
		return gameDifficulty;
	}
}

File: gauguin-core/src/main/kotlin/org/piepmeyer/gauguin/options/DifficultySetting.kt
package org.piepmeyer.gauguin.options

enum class DifficultySetting {
    VERY_EASY,
    EASY,
    MEDIUM,
    HARD,
    EXTREME, ;

    companion object {
        fun all(): Set<DifficultySetting> = entries.toSet()
    }
}
